core/rid.h

```diff
@@ -78,7 +78,16 @@ class RID {
 	}
 	_FORCE_INLINE_ bool is_valid() const { return _data != nullptr; }
 
-	_FORCE_INLINE_ uint32_t get_id() const { return _data ? _data->get_id() : 0; }
+	_FORCE_INLINE_ uint32_t get_id() const {
+		if (_data) {
+			try {
+				return _data->get_id();
+			} catch (...) {
+				ERR_FAIL_V_MSG(0, "RID points to memory that has already been freed.");
+			}
+		}
+		return 0;	// An invalid or uninitialized state
+	}
```

Comment:
Is it necessary for me to `try` here? Or are exceptions handled elsewhere? Since gdscript can still create dangling pointers, it's possible get_id() will reference invalid memory. I tried freeing an RID, then getting it's ID in gdscript, in another build of the engine. I got a non-zero number, but no crash or error message.